### ISSUE TITLE
Potential Feature? Making some of the keyboard more customizable.

### ISSUE THREAD

**Issue by lindoran** [fbergama/pigfx#37]
Since discovering that certain keyboards behave differently with the interface, how hard would it be to make the keyboard  settings available in the .ini file.  things like manual key repeat through software (for keyboards that do not support this behavior), de bounce delays, repeat behavior or general key reassignments for odd keyboards (not just the backspace)?  Just a thought i was having, it should not be considered in any way a high priority. Might also be good to make it posible to switch all that functionality off if everything is working normal..

**Comment by chregu82** [comment]
USB keyboards never do key repeat on their own. It's always done in software. PS/2 keyboard on the other hand do key repeat on their own. If I had a non repeating USB keyboard, I probably could make it working. Something must be different from keyboard to keyboard.

**Comment by lindoran** [comment]
I have several I could send you one if you would like for testing I would
pay the shipping

**Comment by lindoran** [comment]
I I might also be able to run USB debugging and read information from the
keyboard that's not working and compare it to the one that is would that be
helpful?

**Comment by chregu82** [comment]
Let's try option 2 first. This is a test version which outputs the raw keypress data to the screen. I would expect that on key press a hex number >0 is displayed, and on key release 00. Modifier keys like control or alt should display 00 too.
[https://1drv.ms/u/s!Anft_sBKDg5arwr47cL0xuunUqld?e=TIUav4](url)
Image is for Pi1 / Zero only.

**Comment by lindoran** [comment]
the link doesn't appear to be working

**Comment by chregu82** [comment]
I'm sorry, please try again.

**Comment by lindoran** [comment]
ok so here's what happens ... on all the keyboards that don't support autorepeat. they continually send key release 00 unless you press a key then they continually send the hex code for that key... Ill send a video in a few minutes i have to upload it
the keyboard that works is exactly as you described.

**Comment by lindoran** [comment]
https://drive.google.com/file/d/1whNoy8TbhnPC_fQVS_o2-RFWYeuaPC26/view?usp=sharing

**Comment by chregu82** [comment]
I don't seem to have access to this file.

**Comment by lindoran** [comment]
Sorry try this

https://drive.google.com/file/d/1whNoy8TbhnPC_fQVS_o2-RFWYeuaPC26/view?usp=drivesdk


https://drive.google.com/file/d/1whNoy8TbhnPC_fQVS_o2-RFWYeuaPC26/view?usp=drivesdk

**Comment by lindoran** [comment]
I also found that a 00 is sent when releasing a key holding a modifier down
and releasing that modifier. So if you tap a key 00 is sent much faster...

**Comment by chregu82** [comment]
Please try if this works for all your keyboard: [img](https://1drv.ms/u/s!Anft_sBKDg5arwr47cL0xuunUqld?e=TFUZCN)

**Comment by lindoran** [comment]
tested and is working :) actually a bit better than the other keyboard seams far more responsive
thank you so much! its probably fine to push to current.  I tried out the modifier keys as well and caps lock everything works.
you work so fast! I wish I had your stamina.  I got a bit more done on my sprite editor tonight, but its like 1:20 in the morning where I am, im headed to bed. thanks again, much appreciated!
